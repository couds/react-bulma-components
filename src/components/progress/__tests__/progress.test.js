import React from 'react';
import renderer from 'react-test-renderer';
import { Progress } from '..';
import CONSTANTS from '../../../constants';

describe('Progress component', () => {
  it('should exist', () => {
    expect(Progress).toMatchSnapshot();
  });
  it('should have notification classname', () => {
    const component = renderer.create(<Progress value={30} max={100} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const component = renderer.create(<Progress value={30} max={100} className="other-class this-is-a-test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const component = renderer.create(<Progress value={30} max={100} style={{ width: 250 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should be Large', () => {
    const component = renderer.create(<Progress value={30} max={100} size="large" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  Object.values(CONSTANTS.COLORS).map(color =>
    it(`Should use use color ${color}`, () => {
      const component = renderer.create(<Progress value={30} max={100} color={color} />);
      expect(component.toJSON()).toMatchSnapshot();
    })
  );
});
