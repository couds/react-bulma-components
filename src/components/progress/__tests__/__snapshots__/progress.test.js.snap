// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component Should use use color black 1`] = `
<progress
  className="progress is-black"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color danger 1`] = `
<progress
  className="progress is-danger"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color dark 1`] = `
<progress
  className="progress is-dark"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color info 1`] = `
<progress
  className="progress is-info"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color light 1`] = `
<progress
  className="progress is-light"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color link 1`] = `
<progress
  className="progress is-link"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color null 1`] = `
<progress
  className="progress"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color primary 1`] = `
<progress
  className="progress is-primary"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color success 1`] = `
<progress
  className="progress is-success"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color warning 1`] = `
<progress
  className="progress is-warning"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component Should use use color white 1`] = `
<progress
  className="progress is-white"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component should be Large 1`] = `
<progress
  className="progress is-large"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component should concat classname in props with classname 1`] = `
<progress
  className="progress other-class this-is-a-test"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component should exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "backgroundColor": null,
    "className": "",
    "clearfix": false,
    "clipped": false,
    "color": null,
    "hidden": false,
    "invisible": false,
    "italic": false,
    "marginless": false,
    "overlay": false,
    "paddingless": false,
    "pull": null,
    "radiusless": false,
    "responsive": undefined,
    "shadowless": false,
    "size": null,
    "style": Object {},
    "textAlignment": null,
    "textColor": null,
    "textSize": null,
    "textTransform": null,
    "textWeight": null,
    "unselectable": false,
  },
  "propTypes": Object {
    "backgroundColor": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "color": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "max": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "size": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
    "value": [Function],
  },
  "render": [Function],
}
`;

exports[`Progress component should have notification classname 1`] = `
<progress
  className="progress"
  max={100}
  style={Object {}}
  value={30}
/>
`;

exports[`Progress component should use inline styles 1`] = `
<progress
  className="progress"
  max={100}
  style={
    Object {
      "width": 250,
    }
  }
  value={30}
/>
`;
